clp_ffi_py/query_builder.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import List, Optional
+
+from clp_ffi_py.ir import Query
+from clp_ffi_py.wildcard_query import WildcardQuery
+
+
+class QueryBuilderException(Exception):
+    """
+    This exception will be raised when illegal parameters are used to build a
+    Query object.
+    """
+
+    pass
+
+
+class QueryBuilder:
+    """
+    This class serves as an interface for conveniently constructing Query
+    objects utilized in CLP IR streaming search. It provides methods for.
```

Comment:
Details about what parameters can be set feel unnecessary; users can just look at the provided methods. As for details about the syntax of the parameters, I think we can provide very basic details and refer users to the Query docs since that should be the ground truth.